port/common/omrfilestream.c

```diff
@@ -0,0 +1,488 @@
+ *
+ * This file implements file I/O using streams.  It is similar in API to omrfile, and functionally
+ * similar to C standard IO.
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stdio.h>
+#include <unistd.h>
+#include "omriconvhelpers.h"
+#include "omrport.h"
+#include "omrstdarg.h"
+#include "portnls.h"
+#include "ut_omrport.h"
+
+
+#if defined(J9ZOS390)
+/* Needed for a translation table from ascii -> ebcdic */
+#include "atoe.h"
+
+/*
+ * a2e overrides many functions to use ASCII strings.
+ * We need the native EBCDIC string
+ */
+#if defined(fwrite)
+#undef fwrite
+#endif
+#if defined(fread)
+#undef fread
+#endif
+
+#endif /* defined(J9ZOS390) */
+
+/**
+ * @internal
+ * Take a portable buffering mode, and turn it into a platform-specific buffering mode.
+ *
+ * @param[in] mode Portable buffering mode. See @ref omrfilestream_setbuffer for more mode information.
+ * @return Platform specific buffering mode
+ *
+ * @note This will not check to make sure that the mode is valid
+ */
+static int
+translateBufferingMode(int32_t mode)
+{
+	/* On most systems the modes are already mapped correctly */
+	return (int) mode;
+}
+
+/**
+ * @internal
+ * Turns portable open flags into system-specific open flags. These flags will be suitable for the
+ * function fdopen. fdopen does not accept 'b', and does not truncate a file when using 'w' or 'w+'.
+ *
+ * Most flags are ignored by this function: EsOpenCreate, EsOpenCreateAlways, EsOpenCreateNew,
+ * EsOpenTrunc, EsOpenText, EsOpenSync, EsOpenForInherit.
+ *
+ * @param[in] flags The open flags for the functions.  At least one of the reading or writing open
+ * flags should be used: EsOpenWrite, EsOpenRead.
+ *
+ * @return String flags for the filestream. NULL if the flags do not open the file for reading or
+ * writing.
+ */
+static const char *
+EsTranslateOpenFlags(int32_t flags)
+{
+	/* Ignore all other flags: they are taken care of earlier */
+	flags &= (EsOpenWrite | EsOpenRead | EsOpenAppend);
+
+#if defined(J9ZOS390)
+	/* On Z/OS, stdio functions are shadowed to take ASCII strings, and turn them into EBCDIC.
+	 * Since fdopen is a POSIX function, it needs EBCDIC strings.
+	 */
+#pragma convlit(suspend)
+#endif /* defined(J9ZOS390) */
+
+	switch (flags) {
+	case (EsOpenWrite):
+		return "w";
+	case (EsOpenRead):
+		return "r";
+	case (EsOpenWrite | EsOpenRead):
+		return "w+";
+	case (EsOpenAppend | EsOpenWrite):
+		return "a";
+	case (EsOpenAppend | EsOpenRead):
+		return "r";
+	case (EsOpenAppend | EsOpenWrite | EsOpenRead):
+		return "a+";
+	default:
+		return NULL;
+	}
+
+#if defined(J9ZOS390)
+#pragma convlit(resume)
+#endif /* defined(J9ZOS390) */
+
+}
+
+/**
+ * @internal
+ * Determines the proper portable error code to return given a native error code.
+ *
+ * @param[in] errorCode The error code reported by the OS
+ *
+ * @return the (negative) portable error code
+ */
+static int32_t
+findError (int32_t errorCode)
+{
+	switch (errorCode)
```

Comment:
Please follow coding standard for formatting switch statements: https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#formatting-switch-statements